File: revision/saldo_gap.py
#!/usr/bin/env python
# -*- coding: utf8 -*- 

from core.sw_base import BaseHandler


import os
from settings import ROOT_DIR


from tornado import template
import cx_Oracle


from core.common_tools import fetchall, fetchall_by_count

#, fetchone, fetchall_by_name

from avtuk.avtuk_models import RC
#===  =========================================================
class Saldo_GapHandler(BaseHandler):
    urls = r'/revision/auditor'

    def get(self):

        loader = template.Loader(os.path.join(ROOT_DIR, 'revision'))
        result = loader.load("saldo_gap.js").generate()
       
        all_departs = RC.get_current(rc=self.session.rc).depart_cls
        # кнопки с департаментами РЦ
        
        departs = [{"id":"%s" % i.id, "type":"button", "text":i.name, "action":"change_depart"} for i in all_departs]
       
        self.write({'def':[
                           #{'type':"button", 'text':'Печать', 'img':"print.gif", 'imgdis':"print_dis.gif", 'action':'do_print'},
                           #{'type':"separator"},

                           {'type':"button", 'text':'Excel (кратко)',    'action':'do_csv', 'img':"excel24.png", 'imgdis':"excel24g.png"},
                           {'type':"button", 'text':'Excel (подробно)',  'action':'do_csv_full', 'img':"excel24.png", 'imgdis':"excel24g.png"},
                           {'type':"separator"},
                           {'type':"text", 'text':"Подразделение:"},
                           {'id':'departs', 'type':"buttonSelect",  'mode':'select', 'selected': all_departs[0].id,  'items':departs},

                           '''<table style='width:100%'>

                                <tr><td colspan="2"><div id='sw_grid1' style="width:100%; height:250px;"></div></td></tr>
                                <tr><td colspan="2"><span>Детализация:</span></td></tr>
                                <tr><td colspan="2"><div id='sw_grid2' style="width:100%; height:300px;"></div></td></tr>

                              </table>
                           '''],

                    'cmd': result
         
            })


#==============================================================================
class Saldo_GapDataHandler(BaseHandler):
    urls = r'/revision/saldo_gap/data/([^/]+)'


    def get(self, param):
        
        departs = self.get_argument("departs", None)
        code    = self.get_argument("code", None)
            
            
        if param == 'list':
                
            
            out = self.cursor.var(cx_Oracle.CURSOR) 
            
            res = self.proc("shiva.GetSaldoGap", [departs, out])  
            
            all_results = fetchall(res[-1])
            
            self.write(all_results)  
            return



        if param == 'csv':


            out = self.cursor.var(cx_Oracle.CURSOR)

            res = self.proc("shiva.GetSaldoGap", [departs, out])

            all_results = fetchall_by_count(res[-1])

            self.write(all_results)
            return



        if param == 'csv_full':


            out = self.cursor.var(cx_Oracle.CURSOR)

            res = self.proc("shiva.dif_saldo_tehno_shiva", [departs, out])

            all_results = fetchall_by_count(res[-1])

            self.write(all_results)
            return




        if param == 'detail':


            out = self.cursor.var(cx_Oracle.CURSOR)

            res = self.proc("shiva.tovar_address", [departs, code, out])

            all_results = fetchall(res[-1])

            self.write(all_results)
            return

File: revision/saldo_gap.js
  
var sw_grid1 = new dhtmlXGridObject({
   parent: "sw_grid1",
   columns: [
      { type:"ro", sort:"str",  align:"center", width:"80",  label:["Код", "#text_filter"], },  
      { type:"ro", sort:"str",  align:"left"  , width:"*",   label:["Наименование", "#text_filter"] },
      { type:"ro", sort:"int",  align:"center", width:"100", label:"Товарный остаток"},
      { type:"ro", sort:"int",  align:"center", width:"100", label:"Партионный остаток"},                                                   
      { type:"ro", sort:"int",  align:"center", width:"100", label:"Остаток в ячейках"},
      
        ]
    });



var sw_grid2 = new dhtmlXGridObject({
    parent: "sw_grid2",
    columns: [
        { type:"ro", sort:"str",  align:"center", width:"80",  label:"Адрес" },
        { type:"ro", sort:"str",  align:"center", width:"100", label:"Партия" },
        { type:"ro", sort:"str",  align:"center", width:"100", label:"Годен ДО"},
        { type:"ro", sort:"str",  align:"center", width:"100", label:"Количество"},

    ]
});



window.Cleaner.push(sw_grid2);


update = function(){ 
	
	params = {};
	params["departs"]  = self.Toolbars["def"].getListOptionSelected("departs");
	
	
	self.load(sw_grid1, "/revision/saldo_gap/data/list?" + self.serialize(params), function() {
    	
		
    });
};



window.change_depart = function(){

    update();

}

update();



sw_grid1.attachEvent("onRowSelect", function(code){


    params = {"code": code};
    params["departs"]  = self.Toolbars["def"].getListOptionSelected("departs");


    self.load(sw_grid2, "/revision/saldo_gap/data/detail?" + self.serialize(params));

});



window.do_print = function(){

    sw_grid1.printView("","<script> window.print(); </script>");
}



window.do_csv = function(){

    var params = {};

    params["departs"]  = self.Toolbars["def"].getListOptionSelected("departs");


    if (params["departs"] != '3' && params["departs"] != '31'){
        return false;
    }

    var new_grid = new dhtmlXGridObject({
        /*parent: "sw_grid1",*/
        columns: [
            { type:"ro", sort:"str",  align:"center", width:"80",  label:"Код" },
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Наименование" },
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Товарный остаток"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Адрес"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Остаток в ячейке"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Всего на адресах"}

        ]
    });


    self.load(new_grid, "/revision/saldo_gap/data/csv?" + self.serialize(params), function() {

        self.GridCSV(new_grid);
    });


}




window.do_csv_full = function(){

    var params = {};

    params["departs"]  = self.Toolbars["def"].getListOptionSelected("departs");


    if (params["departs"] != '3' && params["departs"] != '31'){
        return false;
    }

    var new_grid = new dhtmlXGridObject({
        /*parent: "sw_grid1",*/
        columns: [
            { type:"ro", sort:"str",  align:"center", width:"80",  label:"Код" },
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Наименование" },
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Товарный остаток"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Адрес"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Остаток в ячейке"},
            { type:"ro", sort:"str",  align:"center", width:"100", label:"Всего на адресах"}

        ]
    });


    self.load(new_grid, "/revision/saldo_gap/data/csv_full?" + self.serialize(params), function() {

        self.GridCSV(new_grid);
    });


}